File: Day_28/Q112.cpp
#include <bits/stdc++.h>
using namespace std;

//Structure For Contact Details
struct Contact{
    string name;
    long long phone;
};

//Add New Contact
void addContact(vector<Contact> &c){

    Contact x;

    cin.ignore();

    cout << "Enter Name : ";
    getline(cin,x.name);

    cout << "Enter Phone Number : ";
    cin >> x.phone;

    c.push_back(x);

    cout << "Contact Added Successfully\n";
}

//Display All Contacts
void displayContacts(vector<Contact> c){

    if(c.size() == 0){
        cout << "No Contact Available\n";
        return;
    }

    for(int i = 0;i < c.size();i++){

        cout << "\nName : " << c[i].name << endl;
        cout << "Phone Number : " << c[i].phone << endl;
    }
}

//Search Contact By Name
void searchContact(vector<Contact> c){

    string name;
    bool found = false;

    cin.ignore();

    cout << "Enter Name : ";
    getline(cin,name);

    for(int i = 0;i < c.size();i++){

        if(c[i].name == name){

            cout << "\nContact Found\n";
            cout << "Phone Number : " << c[i].phone << endl;

            found = true;
        }
    }

    if(found == false) cout << "Contact Not Found\n";
}

int main(){

    vector<Contact> c;
    int choice;

    do{

        //Menu Driven Program
        cout << "\n1. Add Contact";
        cout << "\n2. Display Contacts";
        cout << "\n3. Search Contact";
        cout << "\n4. Exit";
        cout << "\nEnter Choice : ";
        cin >> choice;

        if(choice == 1) addContact(c);

        else if(choice == 2) displayContacts(c);

        else if(choice == 3) searchContact(c);

    }while(choice != 4);

    return 0;
}

File: Day_28/Q112.java
import java.util.*;

//Class For Contact Details
class Contact{
    String name;
    long phone;
}

class Q112{

    static Scanner sc = new Scanner(System.in);

    //Add New Contact
    static void addContact(ArrayList<Contact> c){

        Contact x = new Contact();

        sc.nextLine();

        System.out.print("Enter Name : ");
        x.name = sc.nextLine();

        System.out.print("Enter Phone Number : ");
        x.phone = sc.nextLong();

        c.add(x);

        System.out.println("Contact Added Successfully");
    }

    //Display All Contacts
    static void displayContacts(ArrayList<Contact> c){

        if(c.size() == 0){
            System.out.println("No Contact Available");
            return;
        }

        for(int i = 0;i < c.size();i++){

            System.out.println("\nName : " + c.get(i).name);
            System.out.println("Phone Number : " + c.get(i).phone);
        }
    }

    //Search Contact By Name
    static void searchContact(ArrayList<Contact> c){

        String name;
        boolean found = false;

        sc.nextLine();

        System.out.print("Enter Name : ");
        name = sc.nextLine();

        for(int i = 0;i < c.size();i++){

            if(c.get(i).name.equals(name)){

                System.out.println("\nContact Found");
                System.out.println("Phone Number : " + c.get(i).phone);

                found = true;
            }
        }

        if(found == false) System.out.println("Contact Not Found");
    }

    public static void main(String args[]){

        ArrayList<Contact> c = new ArrayList<>();

        int choice;

        do{

            //Menu Driven Program
            System.out.println("\n1. Add Contact");
            System.out.println("2. Display Contacts");
            System.out.println("3. Search Contact");
            System.out.println("4. Exit");

            System.out.print("Enter Choice : ");
            choice = sc.nextInt();

            if(choice == 1) addContact(c);

            else if(choice == 2) displayContacts(c);

            else if(choice == 3) searchContact(c);

        }while(choice != 4);

        sc.close();
    }
}
